Testbed/Tests/SVQA/Scene2/Scene2Simulation.h

```diff
@@ -0,0 +1,143 @@
+			while(distinc_colors.size() < 4) {
+				SimulationObject::Color c = SimulationObject::getRandomColor();
+				if (std::find(distinc_colors.begin(), distinc_colors.end(), c) == distinc_colors.end()) {
+					distinc_colors.push_back(c);
+				}
+			}
+			
+
+
+			// ------- Stack --------
+			std::vector<SimulationObject::Color> stack_colors;
+			while (stack_colors.size() < 6) {
+				SimulationObject::Color c = SimulationObject::getRandomColor();
+				if (std::find(stack_colors.begin(), stack_colors.end(), c) == stack_colors.end()) {
+					stack_colors.push_back(c);
+				}
+			}
+
+			// bottom left - mid - right
+			AddDynamicObject(b2Vec2(3.0f, 20.0f),b2Vec2(0.0f, 0.0f),SimulationObject::Shape::CUBE, stack_colors[0],SimulationObject::Size::SMALL);
+			AddDynamicObject(b2Vec2(5.2f, 20.0f),b2Vec2(0.0f, 0.0f),SimulationObject::Shape::CUBE, stack_colors[1],SimulationObject::Size::SMALL);
+			AddDynamicObject(b2Vec2(7.4f, 20.0f),b2Vec2(0.0f, 0.0f),SimulationObject::Shape::CUBE, stack_colors[2],SimulationObject::Size::SMALL);
+			// middle left - right
+			AddDynamicObject(b2Vec2(4.0f, 22.0f), b2Vec2(0.0f, 0.0f), SimulationObject::Shape::CUBE, stack_colors[3], SimulationObject::Size::SMALL);
+			AddDynamicObject(b2Vec2(6.2f, 22.0f), b2Vec2(0.0f, 0.0f), SimulationObject::Shape::CUBE, stack_colors[4], SimulationObject::Size::SMALL);
+			// top
+			AddDynamicObject(b2Vec2(5.0f, 24.0f), b2Vec2(0.0f, 0.0f), SimulationObject::Shape::CUBE, stack_colors[5], SimulationObject::Size::SMALL);
+
+
+
+
+			// ------- Top Right Platform and Circle -------
+			AddStaticObject(b2Vec2(17.0f, 19.0f), 0, SimulationObject::STATIC_PLATFORM);
+			AddStaticObject(b2Vec2(9.0f, 19.0f), 0, SimulationObject::STATIC_PLATFORM);
+			AddStaticObject(b2Vec2(11.0f, 19.0f), 0, SimulationObject::STATIC_RAMP);
+
+			float32 tr_circle_velocities[5] = { -34.0f, -28.0f, -20.0f, -17.0f, -14.0f };
+			
+			AddDynamicObject(
+				b2Vec2(22.0f, 20.0f),
+				b2Vec2(tr_circle_velocities[(int)RandomFloatFromHardware(0.0, 5.0)], 0.0f),
+				SimulationObject::Shape::CIRCLE,
+				distinc_colors[0],
+				SimulationObject::getRandomSize()
+			);
+
+
+			// ------- Inclined Platform and Rolling Circle -------
+			AddStaticObject(b2Vec2(-26.5f, 12.0f), 6* M_PI / 7, SimulationObject::STATIC_PLATFORM);
+			float32 rolling_circle_heights[] = { 29.0f, 18.0f, 25.0f, 40.0f, 19.0f };
+
+			AddDynamicObject(
+				b2Vec2(-22.0f, rolling_circle_heights[(int)RandomFloatFromHardware(0.0, 5.0)]),
+				b2Vec2(0.0f, 0.0f),
+				SimulationObject::Shape::CIRCLE,
+				distinc_colors[1],
+				SimulationObject::getRandomSize()
+			);
+
+
+
+			// ------- Bottom Left Platform and Circle -------
+			AddStaticObject(b2Vec2(-7.5f, 6.0f), 0, SimulationObject::STATIC_PLATFORM);
+
+			AddDynamicObject(
+				b2Vec2(-9.0f, 8.0f),
+				b2Vec2(0.0f, 0.0f),
+				SimulationObject::Shape::CIRCLE,
+				distinc_colors[2],
+				SimulationObject::getRandomSize()
+			);
+
+
+
+
+			// Basket
+			AddTargetBasket(b2Vec2(4.5f, -0.90f), 0.0f);
```

Comment:
We are adding a basket after we added all the other objects. I think that's the reason behind the issue #40.